File: scripts/clear_descriptions.py
import os
from pathlib import Path
import re
from tqdm import tqdm

class DescriptionCleaner:
    def __init__(self):
        self.base_path = Path("pages/espanol")
        self.report = {
            'total_files': 0,
            'files_processed': 0,
            'descriptions_removed': 0,
            'errors': []
        }

    def find_exercise_files(self):
        """Find all exercise files in the Spanish directory."""
        self.exercise_files = []
        
        # Walk through all directories
        for week_dir in self.base_path.glob("semana[1-5]"):
            for day_dir in week_dir.glob("dia[1-5]"):
                for exercise_file in day_dir.glob("ejercicio[1-5].js"):
                    self.exercise_files.append(exercise_file)
                    self.report['total_files'] += 1

    def process_file(self, file_path):
        """Process a single exercise file to remove descriptions."""
        try:
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()

            # Find all EspDrillNavBar components
            nav_bars = re.finditer(r'<EspDrillNavBar.*?>', content)
            
            # Process each nav bar
            for match in nav_bars:
                nav_bar = match.group(0)
                
                # Handle multiple description attributes
                desc_parts = re.split(r'(description="[^"]+")', nav_bar)
                cleaned_parts = []
                for part in desc_parts:
                    if part.startswith('description="'):
                        self.report['descriptions_removed'] += 1
                    else:
                        cleaned_parts.append(part)
                cleaned_nav = ''.join(cleaned_parts)
                
                # Replace the old nav bar with the cleaned version
                content = content.replace(match.group(0), cleaned_nav)

            # Also check for description in EspDrillBarTemplate
            template_match = re.search(r'<EspDrillBarTemplate.*?>', content)
            if template_match:
                template = template_match.group(0)
                desc_parts = re.split(r'(description="[^"]+")', template)
                cleaned_parts = []
                for part in desc_parts:
                    if part.startswith('description="'):
                        self.report['descriptions_removed'] += 1
                    else:
                        cleaned_parts.append(part)
                cleaned_template = ''.join(cleaned_parts)
                
                # Replace the old template with the cleaned version
                content = content.replace(template_match.group(0), cleaned_template)

            # Write changes back to file if any were made
            if self.report['descriptions_removed'] > 0:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(content)
                self.report['files_processed'] += 1

        except Exception as e:
            self.report['errors'].append(f"Error processing {file_path}: {str(e)}")

    def generate_report(self):
        """Generate a report of the cleaning process."""
        print("\n=== CLEANING REPORT ===")
        print(f"Total files: {self.report['total_files']}")
        print(f"Files processed: {self.report['files_processed']}")
        print(f"Descriptions removed: {self.report['descriptions_removed']}")
        print(f"Errors: {len(self.report['errors'])}")

        if self.report['errors']:
            print("\n=== ERRORS ===")
            for error in self.report['errors']:
                print(f"- {error}")

    def run(self):
        """Run the cleaning process."""
        print("\nStarting description cleanup...")
        
        self.find_exercise_files()
        
        for file_path in tqdm(self.exercise_files, desc="Processing files", unit="file"):
            self.process_file(file_path)

        self.generate_report()

if __name__ == "__main__":
    cleaner = DescriptionCleaner()
    cleaner.run()

File: scripts/clear-descriptions.js
const fs = require('fs');
const path = require('path');
const parser = require('@babel/parser');
const traverse = require('@babel/traverse').default;
const generate = require('@babel/generator').default;
import ora from 'ora';

const report = {
    totalFiles: 0,
    filesProcessed: 0,
    descriptionsRemoved: 0,
    errors: []
};

// Function to process a single file
async function processFile(filePath) {
    try {
        const content = fs.readFileSync(filePath, 'utf-8');
        const ast = parser.parse(content, {
            sourceType: 'module',
            plugins: ['jsx']
        });

        // Traverse the AST to find and modify the components
        traverse(ast, {
            JSXOpeningElement(path) {
                const componentName = path.node.name.name;
                
                if (componentName === 'EspDrillNavBar' || componentName === 'EspDrillBarTemplate') {
                    const originalAttributes = path.node.attributes;
                    
                    // Filter out description attributes
                    path.node.attributes = path.node.attributes.filter(
                        attr => attr.name && attr.name.name !== 'description'
                    );
                    
                    // Count removed descriptions
                    const descriptionsRemoved = originalAttributes.length - path.node.attributes.length;
                    report.descriptionsRemoved += descriptionsRemoved;
                }
            }
        });

        // Generate the modified code
        const output = generate(ast, { retainLines: true }).code;
        
        // Only write if changes were made
        if (report.descriptionsRemoved > 0) {
            fs.writeFileSync(filePath, output, 'utf-8');
            report.filesProcessed++;
        }
    } catch (error) {
        report.errors.push(`Error processing ${filePath}: ${error.message}`);
    }
}

// Function to find all exercise files
async function findExerciseFiles() {
    const exerciseFiles = [];
    const baseDir = path.join(__dirname, '../pages/espanol');

    // Walk through all directories
    for (let week = 1; week <= 5; week++) {
        for (let day = 1; day <= 5; day++) {
            const weekDir = path.join(baseDir, `semana${week}`);
            const dayDir = path.join(weekDir, `dia${day}`);
            
            if (fs.existsSync(dayDir)) {
                const files = fs.readdirSync(dayDir);
                for (const file of files) {
                    if (file.startsWith('ejercicio') && file.endsWith('.js')) {
                        exerciseFiles.push(path.join(dayDir, file));
                        report.totalFiles++;
                    }
                }
            }
        }
    }

    return exerciseFiles;
}

// Main function
async function main() {
    const spinner = ora('Finding exercise files...').start();
    const exerciseFiles = await findExerciseFiles();
    spinner.succeed(`Found ${report.totalFiles} exercise files`);

    spinner.start('Processing files...');
    for (const file of exerciseFiles) {
        await processFile(file);
    }
    spinner.succeed('Processing complete');

    // Generate report
    console.log('\n=== CLEANING REPORT ===');
    console.log(`Total files: ${report.totalFiles}`);
    console.log(`Files processed: ${report.filesProcessed}`);
    console.log(`Descriptions removed: ${report.descriptionsRemoved}`);
    console.log(`Errors: ${report.errors.length}`);

    if (report.errors.length > 0) {
        console.log('\n=== ERRORS ===');
        report.errors.forEach(error => console.log(`- ${error}`));
    }
}

// Run the script
main().catch(error => {
    console.error('Script failed:', error);
    process.exit(1);
});
